src/platform/packages/shared/kbn-unified-tabs/src/components/tabbed_content/tabbed_content.tsx

```diff
@@ -110,27 +156,57 @@ export const TabbedContent: React.FC<TabbedContentProps> = ({
         if (nextState.selectedItem) {
           tabsBarApi.current?.moveFocusToNextSelectedItem(nextState.selectedItem);
         }
+
+        onEvent('tabClosed', {
+          totalTabsOpen: prevState.items.length,
+          remainingTabsCount: nextState.items.length,
+          tabId: item.id,
+        });
+
         return nextState;
       });
     },
-    [changeState]
+    [changeState, onEvent]
   );
 
   const onReorder = useCallback(
     (reorderedItems: TabItem[]) => {
-      changeState((prevState) => ({
-        ...prevState,
-        items: reorderedItems,
-      }));
+      changeState((prevState) => {
+        const prevItems = prevState.items;
+        const movedItem = prevItems.find((item, index) => item.id !== reorderedItems[index]?.id);
```

Comment:
Will this work reliably? Seems like if I had tabs `A, B, C`, then dragged C before B like `A, C, B`, it would return B instead of C. Is there a way we can access the actual tab that triggered the event instead?

Reply:
Ahh you're right. I'll rethink that